### PR TITLE
Playlists branch

### PR DESCRIPTION
**Pull request by tejas-muthya37**
This PR contains code for the Playlists branch.

The code to be reviewed is present in src -> Screens -> Playlists, src -> Screens -> Playlist, src -> Components -> PlaylistThumbnail and src -> Context -> products-context.js

### CODE REVIEW

File: src/Context/products-context.js [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        .catch((err) => console.log(err));
+      return {
+        ...state,
+        likedArray: state.likedArray.filter(
+          (movie) => movie._id !== action.payload._id
+        ),
+        dislikedArray: [...state.dislikedArray, action.payload],
+      };
+    case "Add to Watch Later":
+      const watchLaterVideoFound = state.watchLaterArray.find(
```
Comment (yaswanthmyneni) [inline src/Context/products-context.js:216]: changes: write case names as 'ADD_TO_WATCH_LATER'

File: src/Context/products-context.js [hunk 2]
```diff
@@ -0,0 +1,348 @@
+import { useReducer, useContext, createContext } from "react";
+
+const ProductsContext = createContext(null);
+
+const reducer = (state, action) => {
+  const encodedToken = localStorage.getItem("ENCODED_TOKEN_2");
```
Comment (yaswanthmyneni) [inline src/Context/products-context.js]: suggestion: destructure the 'action' in parameters.

File: src/Context/toast-context.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const ToastContext = createContext();
+
+const ToastProvider = ({ children }) => {
+  const [toastVisibility, setToastVisibility] = useState("hidden");
+  const [toastText, setToastText] = useState("");
+  const [toastColor, setToastColor] = useState({
+    color: "",
+    backgroundColor: "",
+  });
```
Comment (yaswanthmyneni) [inline src/Context/toast-context.js:11]: changes: use 'useReducer' instead of 'useState'.

File: src/Screens/Playlist/Playlist.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                playlist.videos.map((movie) => {
+                  return (
+                    <Movie
+                      key={movie.id}
+                      id={movie.id}
+                      source={movie.source}
+                      thumbnail={movie.thumbnail}
+                      title={movie.title}
+                      playlistPage={true}
+                      category={movie.category}
```
Comment (yaswanthmyneni) [inline src/Screens/Playlist/Playlist.js:75]: changes: send data like < Movie key={movie.id} movie={movie} / >

File: src/Components/PlaylistThumbnail/PlaylistThumbnail.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import "./playlistThumbnail.css";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { Link } from "react-router-dom";
+
+const PlaylistThumbnail = (props) => {
+  return (
```
Comment (yaswanthmyneni) [inline src/Components/PlaylistThumbnail/PlaylistThumbnail.js:6]: suggestion: destructure props.

**Review by yaswanthmyneni — APPROVED** [review]
you can merge after changes

File: src/Context/products-context.js [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+    }
+  )
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "Add to Liked":
```
Comment (Ronak-Mutha) [inline src/Context/products-context.js:173]: Prefer using upper case while naming action types i.e. 'ADD_TO_LIKED'

**Review by Ronak-Mutha — COMMENTED** [review]
Code looks clean and good to merge, apart from the destructuring and naming related things highlighted by Yaswanth.

File: src/Context/products-context.js (hunk 6 above)
Comment (SJTGSHIVAM) [reply src/Context/products-context.js:173]: @tejas-muthya37 I would encourage you to make constants and then use constants that is even better

File: src/Components/PlaylistThumbnail/playlistThumbnail.css [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    gap: 2rem;
+    text-align: left;
+}
+
+.thumbnail-left>* {
+    text-decoration: none;
+    color: whitesmoke;
+}
+
+.thumbnail-right>* {
```
Comment (SJTGSHIVAM) [inline src/Components/PlaylistThumbnail/playlistThumbnail.css:37]: spacing can make things a lot more reasdable

File: src/Screens/Playlist/playlist.css [hunk 8]
```diff
@@ -0,0 +1,32 @@
+.Playlist {
```
Comment (SJTGSHIVAM) [inline src/Screens/Playlist/playlist.css:1]: capital in css naming is not common

**Review by SJTGSHIVAM — APPROVED** [review]
Reading your code for 2-3 weeks now the quality of your code has increased a good amount. Have mentioned some comments. The code looks okay for now.

File: src/Components/PlaylistThumbnail/playlistThumbnail.css (hunk 7 above)
Comment (tejas-muthya37) [reply src/Components/PlaylistThumbnail/playlistThumbnail.css:37]: Noted, this is VS Code's default formatter doing it's thing.

File: src/Screens/Playlist/playlist.css (hunk 8 above)
Comment (tejas-muthya37) [reply src/Screens/Playlist/playlist.css:1]: Noted.

**Pull request merged**